File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/GetWarehouseUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.result.WarehouseResult;
import reactor.core.publisher.Mono;

public interface GetWarehouseUseCase {

    Mono<WarehouseResult> getById(Long warehouseId);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/GetWarehouseUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.result.WarehouseResult
import reactor.core.publisher.Mono

interface GetWarehouseUseCase {
    fun getById(warehouseId: Long): Mono<WarehouseResult>
}
